scripts/visualize/timeDayUse.R

```diff
@@ -4,88 +4,94 @@ library(ggplot2)
 library(scales)
 
 visualize.portfolio_timeDayUse <- function(viz=as.viz("portfolio_timeDayUse_year")) {
-  
+
   dep_data <- readDepends(viz)
-  
-  viz.data <- dep_data[["viz_data"]] 
+
+  viz.data <- dep_data[["viz_data"]]
   project.data <- dep_data[["project_table"]]
-  
+
   height = viz[["height"]]
   width = viz[["width"]]
   plot_type <- viz[["plottype"]]
   bar_line_col = viz[["bar_line_col"]]
-  
+
   port_device <- plot_tod(viz.data, bar_line_col, project.data)
-  
-  ggsave(port_device, filename = viz[["location"]], 
+
+  ggsave(port_device, filename = viz[["location"]],
          height = height, width = width)
-  
-  
+
+
 }
 
 visualize.timeDayUse_app <- function(viz=as.viz("timeDayUse_app_year")) {
-  
+
   library(dplyr)
   library(ggplot2)
   library(scales)
-  
+
   dep_data <- readDepends(viz)
-  viz.data <- dep_data[["viz_data"]] 
+  viz.data <- dep_data[["viz_data"]]
   project.data <- dep_data[["project_table"]]
-  
+
   height = viz[["height"]]
   width = viz[["width"]]
   plot_type <- viz[["plottype"]]
   bar_line_col = viz[["bar_line_col"]]
-  
+
   x <- data.frame(id = character(),
                   loc = character(),
                   type = character(),
                   stringsAsFactors = FALSE)
-  
+
   for(i in unique(viz.data$viewID)) {
-    
+
     location <- paste0("cache/visualize/", i,plot_type, ".png")
-    
-    sub.data <- filter(viz.data, viewID == i) 
+
+    sub.data <- filter(viz.data, viewID == i)
 
     port_device <- plot_tod(sub.data,bar_line_col, project.data)
-    
-    ggsave(port_device, filename = location, 
+
+    ggsave(port_device, filename = location,
            height = height, width = width)
-    
+
     x <- bind_rows(x, data.frame(id = i,
                                  loc = location,
                                  type = plot_type,
-                                 stringsAsFactors = FALSE))    
+                                 stringsAsFactors = FALSE))
   }
   write.csv(x, file=viz[["location"]], row.names = FALSE)
 }
 
 plot_tod <- function(viz.data, bar_line_col, project.data){
-  
+
   central_sites <- project.data$viewID[project.data$timezone == "America/Chicago"]
-  
+
   viz.data$hour[viz.data$viewID %in% central_sites] <- as.character(as.numeric(viz.data$hour[viz.data$viewID %in% central_sites]) + 1)
-  
-  
-  hourSum <- viz.data %>% 
-    group_by(hour) %>% 
-    summarise(n = n()) #need to set to numeric?
-  
-  hourSum$hour <- as.numeric(hourSum$hour)
-  
+
+  hourSum <- viz.data %>%
+    mutate(hour = as.numeric(hour)) %>%
+    group_by(hour) %>%
+    summarise(Sessions = n())
+
+  # Because of central time conversions, there were 25's in there:
+  hourSum$hour[hourSum$hour > 24] <- hourSum$hour[hourSum$hour > 24] - 24
```

Comment:
@eread-usgs the conversion of the time-of-day happens (already) in line 67. This PR adds a midnight fix because there were some "hour 25"'s being displayed